pkg/prometheus/promcfg.go

```diff
@@ -322,14 +322,29 @@ func honorLabels(userHonorLabels, overrideHonorLabels bool) bool {
 	return userHonorLabels
 }
 
+// honorTimestamps determines the value of honor_timestamps.
+// if overrideHonorTimestamps is true, then we want to always
+// set honor_timestamps to true. Otherwise we set it to whateever
+// user wants, while assuming no setting as true.
+func honorTimestamps(userHonorTimestamps *bool, overrideHonorTimestamps bool) bool {
```

Comment:
I am so confused if `overrideHonorTimestamps` means `allowOverridingHonorTimestamp` or actually `overriddenHonorTimestamp` :see_no_evil: I think the latter.. based on comment.

Reply:
`overrideHonorTimestamps: true` discards whatever was set in ServiceMonitor/PodMonitor and sets `honor_timestamps: true` in each scrape config.